source/plugins/go/src/oms.go

```diff
@@ -1281,42 +1284,77 @@ func PostDataHelper(tailPluginRecords []map[interface{}]interface{}) int {
 			msgpBytes = msgp.AppendMapStrStr(msgpBytes, fluentForward.Entries[entry].Record)
 		}
 
-		if MdsdMsgpUnixSocketClient == nil {
-			Log("Error::mdsd::mdsd connection does not exist. re-connecting ...")
-			CreateMDSDClient(ContainerLogV2, ContainerType)
-			if MdsdMsgpUnixSocketClient == nil {
-				Log("Error::mdsd::Unable to create mdsd client. Please check error log.")
-
+		if IsWindows == true {
+			jsonStr, err := json.Marshal(stringMap)
+			Log("AMA::fluentForwardTag: %s, jsonStr: %s, msgpSize: %d", fluentForward.Tag, jsonStr, msgpSize)
+			if ContainerLogNamedPipe == nil {
+				Log("Error::AMA:: The connection to named pipe was nil. re-connecting...")
+				if ContainerLogSchemaV2 {
+					CreateWindowsNamedPipesClient(extension.GetInstance(FLBLogger, ContainerType).GetOutputNamedPipe(ContainerLogV2DataType))
+				} else {
+					CreateWindowsNamedPipesClient(extension.GetInstance(FLBLogger, ContainerType).GetOutputNamedPipe(ContainerLogDataType))
+				}
```

Comment:
if else can be cleaned up. Get the outputnamedpipe based on schema and then create the actual namedpipe